src/Services/BlazorStaticService.cs

```diff
@@ -77,12 +80,59 @@ internal async Task GenerateStaticPages(string appUrl)
                 Directory.CreateDirectory(directoryPath);
 
             await File.WriteAllTextAsync(outFilePath, content);
+
+        }
+
+        if (Options.ShouldGenerateSitemap)
+            await GenerateSitemap();
+    }
+
+    /// <summary>
+    /// Generates an XML sitemap from the registered URLs. <br />
+    /// !requires BlazorStaticOptions.SiteUrl to not be null!
+    /// </summary>
+    private async Task GenerateSitemap()
+    {
+        // todo: should we throw an error here? show a warning and use localhost?
+        if (string.IsNullOrWhiteSpace(Options.SiteUrl)) return;
+
+        XNamespace xmlns = XNamespace.Get("http://www.sitemaps.org/schemas/sitemap/0.9");
+        List<XElement> xmlUrlList = [];
+
+        foreach (PageToGenerate page in options.PagesToGenerate)
+        {
+            if (!Uri.TryCreate(new Uri(Options.SiteUrl), relativeUri: page.Url, out Uri? pageUrl)) continue;
+
+            List<XElement> xElements = [new XElement(xmlns + "loc", EncodeUrl(pageUrl))];
+
+            // only add a <lastmod> node if the file is a blog post
+            // todo?: should we also check last write time for razor files?
```

Comment:
As described in previous comment, bring the publish date here... Wrapped with some sort of `AdditionalInfo`